frontend/src/app/pages/search/search.component.ts

```diff
@@ -42,14 +45,22 @@ export class SearchComponent {
     ];
 
     this.sortingDirections = [
-      { label: 'Ascending', value: SorintgDirections.Asceding },
-      { label: 'Descending', value: SorintgDirections.Descending }
+      { label: 'Ascending', value: SortingDirections.Asceding },
+      { label: 'Descending', value: SortingDirections.Descending }
     ]
-  }
 
-  ngOnInit() {
     this.sortingDirection = this.sortingDirections[0];
     this.sortingType = this.sortingTypes[0];
+
+    let routerState = this.router.getCurrentNavigation()?.extras.state;
+    if (routerState != undefined) {
+      this.query = routerState['query'];
+      console.log(this.query)
+      this.performSearch();
+    }
+  }
+
+  ngOnInit() {
```

Comment:
idk if its empty it is getting displayed quite weird by github but if it is delete